Feature/orthorex calculator

ebuka-nwabueze opened the pull request.

Ozoemenamsolomon committed:
fix:header bug

ebuka-nwabueze committed:
initial setup

ebuka-nwabueze committed:
initial setup

ebuka-nwabueze committed:
adjust the stylying to match the fixed heading

Ozoemenamsolomon committed:
header works better, I think

- remove extra margin for page layouts
- add weight to product data

ebuka-nwabueze committed:
first form template added

ebuka-nwabueze committed:
add the preview final amount for calculator

ebuka-nwabueze committed:
first part of the design complete

ebuka-nwabueze committed:
fix the length of input

Ozoemenamsolomon commented on src/components/calculator/FinalAmount.tsx:
```diff
@@ -0,0 +1,65 @@
+
+interface FinalAmountType {
+	data: {
+		productType: string;
```
when you see yourself naming an attribute `somethingType` then you should consider using an enum, I think

ebuka-nwabueze replied:
Ok. Used the enum type here.

Ozoemenamsolomon commented on src/components/styled/Forms.tsx:
```diff
@@ -41,3 +41,151 @@ export const StyledFormButtonControl = styled.div`
+		box-shadow: 0 0 0 1px var(--oex-orange);
+	}
+
+	&::checked::before {
```
nice, just change one attribute on state change

ebuka-nwabueze replied:
Could you point out here what you meant?

Ozoemenamsolomon replied:
No attention is needed, just commending

Ozoemenamsolomon commented on src/pages/calculator.tsx:
```diff
@@ -0,0 +1,328 @@
+										<FormRadioLabel>
+											<input
+												type="radio"
+												id="shape"
```
your ids should be unique, compare them with ln100. That's the key for your `onInputChange` above

ebuka-nwabueze replied:
What is In100? Should the id be something like shape100?

ebuka-nwabueze replied:
If the shape has a different id, i will use the name as the key, for radio buttons to work together, they must have the same name attribute

Ozoemenamsolomon replied:
line 100

Ozoemenamsolomon replied:
consider using the name in your onChange then, instead of ID, so that ID would be unique in a page

Ozoemenamsolomon commented on src/pages/calculator.tsx:
```diff
@@ -0,0 +1,328 @@
+	| React.ChangeEvent<HTMLSelectElement>;
+
+type FormDataType = {
+	shape: string;
```
enum too, I guess

ebuka-nwabueze replied:
OK. Changed the type as well to enum.

Ozoemenamsolomon commented on src/pages/calculator.tsx:
```diff
@@ -0,0 +1,328 @@
+
+type FormDataType = {
+	shape: string;
+	diameter: string;
```
it should all be number, width, and length too

ebuka-nwabueze replied:
Ok. Done

Ozoemenamsolomon commented on src/pages/calculator.tsx:
```diff
@@ -0,0 +1,328 @@
+	margin-top: 5rem;
+	padding-top: 7rem;
+
+	@media (min-width: 768px) {
```
start using the styled-component themed media query eg.  `@media ${({theme})=>theme.breakpoints.above.md}`

ebuka-nwabueze replied:
Ok. Done

Ozoemenamsolomon commented on src/pages/composites/index.tsx:
```diff
@@ -72,7 +72,11 @@ const expectancies: ServiceCardType[] = [
 			'Use our resin calculator to estimate the amount of epoxy resin you will need for your projects.',
 		image: calculator,
-		cta: <CTA white>Calculate your resin</CTA>,
+		cta: (
```
nice

Ozoemenamsolomon commented on src/styles/globals.css:
```diff
@@ -15,7 +15,6 @@ body {
 	--oex-lighter-grey: #e5e5e5;
 	--oex-lightest-grey: #f2f0f0;
 	--oex-orange: #fe7624;
-	--oex-light-orange: #fef9f5;
```
some of these variables look alike, they need to be reduced

ebuka-nwabueze replied:
I won't add anymore colors, i will use only these ones. I'm trying to get the design

Ozoemenamsolomon replied:
Alright, consider adding them only if there's a very obvious visual difference from an already existing hex

Ozoemenamsolomon reviewed (commented):
That's all for now

Ozoemenamsolomon commented:
![image](https://user-images.githubusercontent.com/52500221/209661316-e01830f4-a99b-460a-9449-423070d68a39.png)
one of them should be selected by default, which in turn determines what's displayed

ebuka-nwabueze commented:
> ![image](https://user-images.githubusercontent.com/52500221/209661316-e01830f4-a99b-460a-9449-423070d68a39.png) one of them should be selected by default, which in turn determines what's displayed

If one is checked, the value does not change anymore.

ebuka-nwabueze committed:
rectangle calculation created

ebuka-nwabueze committed:
remove unused variable

ebuka-nwabueze committed:
change the breapoint to the styled theme

ebuka-nwabueze committed:
remove unused variable

ebuka-nwabueze committed:
add cylinder calculation

ebuka-nwabueze committed:
add cylinder calculation

ebuka-nwabueze committed:
add the calculate the resin in kg for cylinder shape

Ozoemenamsolomon commented on src/utils/calculator.ts:
```diff
@@ -0,0 +1,99 @@
+const feetToMetre = 0.3048;
+const densityOfResinInKg = 1110;
+
+const convertToMetre = (unit: string) => {
```
The parameter should be oof type `UnitType`

ebuka-nwabueze replied:
Done

Ozoemenamsolomon commented on src/utils/calculator.ts:
```diff
@@ -0,0 +1,99 @@
+import { UNITSTYPE } from 'pages/calculator';
```
please only export the page component and the nextjs data fetrching funtions in the pages directory, every other export should be somewhere else eg. components, utils etc.

ebuka-nwabueze replied:
Ok, I have transfered the defintion to the utils

Ozoemenamsolomon commented on src/pages/calculator.tsx:
```diff
@@ -0,0 +1,386 @@
+	calculateRectangularResinInKg,
+} from 'utils/calculator';
+
+export enum PRODUCTTYPE {
```
Please remove all these types export and take them somewhere else, maybe a separate `data` file or so

Ozoemenamsolomon replied:
`FinalAmount.tsx` would ba a better option

ebuka-nwabueze replied:
Sure done

Ozoemenamsolomon commented on src/components/Header.tsx:
```diff
@@ -33,11 +33,14 @@ const Header: React.FC<HeaderProp> = ({ pathname }) => {
 	const [light, setLight] = useState(false);
 
 	useEffect(() => {
+<<<<<<< HEAD
```
please look into this theonly change should be the addition to calculator route

ebuka-nwabueze replied:
Yeah, i fixed it in the file.

Ozoemenamsolomon reviewed (changes requested):
👍🏽

Ozoemenamsolomon commented on src/pages/calculator.tsx:
```diff
@@ -0,0 +1,386 @@
+	OEX5302 = 'OEX5302',
+	OEX5311 = 'OEX5311',
+}
+export enum SHAPETYPE {
```
You exported thisb but it's not used anywhere else

Ozoemenamsolomon commented on src/pages/calculator.tsx:
```diff
@@ -0,0 +1,386 @@
+	OEX5302 = 'OEX5302',
+	OEX5311 = 'OEX5311',
+}
+export enum SHAPETYPE {
```
You exported this but it's not used anywhere else

ebuka-nwabueze replied:
moved it to the utils file

ebuka-nwabueze committed:
refactor the calculator function

ebuka-nwabueze committed:
reset the calculator function from previous refactored state

ebuka-nwabueze committed:
add radiobutton change to reset diameter, length values

ebuka-nwabueze committed:
update the calculator

ebuka-nwabueze committed:
remove unused variables

ebuka-nwabueze committed:
add selected products and adjust font size for product cardd

ebuka-nwabueze committed:
remove required props for arrow back component

ebuka-nwabueze committed:
remove required props for arrow back component

ebuka-nwabueze committed:
add breadcrumb for calculator desktop view

ebuka-nwabueze committed:
adjust breadcrumb into page container

ebuka-nwabueze committed:
adjust color  of page calculator background

Ozoemenamsolomon commented on src/pages/calculator.tsx:
```diff
@@ -0,0 +1,455 @@
+	}
+`;
+
+export const ProductCards = styled.div`
```
move this to a styled folder, only export the page or the data fetching functions from a page directory.

ebuka-nwabueze replied:
Ok. I won't need the export again. So i removed the export

ebuka-nwabueze committed:
adjust styling

ebuka-nwabueze committed:
adjust styling to the selected card

Ozoemenamsolomon commented:
remember to set the PR as ready, when you think it's ready

ebuka-nwabueze committed:
ädjust the unit selector -- remove the breadcrumbs

ebuka-nwabueze committed:
remove export of product cards

Ozoemenamsolomon committed:
style: move match diff lines

Ozoemenamsolomon reviewed (approved):
LGTM but the components in `src/components/styled/Forms.tsx` need to be looked into

Ozoemenamsolomon committed:
style: unify `ProductCards`
- fix: product card hover shadow not showing
- modify next link migration
- remove extra css variables

Ozoemenamsolomon committed:
fix: mobile categorycards
